qiskit_experiments/framework/experiment_data.py

```diff
@@ -767,6 +770,81 @@ def add_data(
     ) -> None:
         """Add experiment data.
 
+        Args:
+            data: Experiment data to add. Several types are accepted for convenience:
+
+                * Result: Add data from this ``Result`` object.
+                * List[Result]: Add data from the ``Result`` objects.
+                * Dict: Add this data.
+                * List[Dict]: Add this list of data.
+
+        Raises:
+            TypeError: If the input data type is invalid.
+        """
+
+        if any(not future.done() for future in self._analysis_futures.values()):
+            LOG.warning(
+                "Not all analysis has finished running. Adding new data may "
+                "create unexpected analysis results."
+            )
+        if not isinstance(data, list):
+            data = [data]
+
+        # Directly add non-job data
+        with self._result_data.lock:
+            tmp_exp_data = ExperimentData()
+            composite_flag = False
+            experiment_seperator = defaultdict(lambda : ExperimentData())
+            for datum in data:
+                if isinstance(datum, dict):
+                    if "metadata" in datum and "composite_metadata" in datum["metadata"]:
+                        composite_flag = True
+                        experiment_seperator[datum["metadata"]["experiment_type"]].add_data(datum["metadata"]["composite_metadata"])
+                        marginalized_datum = self._marginalized_component_data([datum])
+                        for inner_datum in marginalized_datum:
+                            for inner_inner_datum in inner_datum:
+                                experiment_seperator[datum["metadata"]["experiment_type"]].__add_data([inner_inner_datum])
+                    elif "composite_metadata" in datum:
```

Comment:
When this evaluation returns True? Because `"composite_metadata"` is a part of the metadata, and when the `datum` is just a metadata, there is no count information in the object, and there is nothing to store.